src/Mvc/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs

```diff
@@ -22,200 +46,325 @@ public static int Process(GetDocumentCommandContext context)
+            {
+                return false;
+            }
+
+            var generateMethod = GetMethod(
+                GenerateMethodName,
+                serviceType,
+                GenerateMethodParameterTypes,
+                GenerateMethodReturnType,
+                reportErrors: true,
+                isStatic: false);
+            if (generateMethod == null)
+            {
+                return false;
+            }
+
+            var service = services.GetService(serviceType);
+            if (service == null)
+            {
+                Reporter.WriteError(Resources.FormatServiceNotFound(DocumentService));
+                return false;
+            }
+
+            var documentNames = (IEnumerable<string>)InvokeMethod(getDocumentsMethod, service, GetDocumentsArguments);
+            if (documentNames == null)
+            {
+                return false;
+            }
+
+            // Write out the documents.
+            Directory.CreateDirectory(context.OutputDirectory);
+            var filePathList = new List<string>();
+            foreach (var documentName in documentNames)
+            {
+                var filePath = GetDocument(
+                    documentName,
+                    context.ProjectName,
+                    context.OutputDirectory,
+                    generateMethod,
+                    service);
+                if (filePath == null)
                 {
-                    Reporter.WriteWarning(Resources.FormatMethodReturnTypeUnsupported(
-                        methodName,
-                        serviceName,
-                        method.ReturnType,
-                        typeof(Task)));
                     return false;
                 }
 
-                var service = services.GetService(serviceType);
-                if (service == null)
+                filePathList.Add(filePath);
+            }
+
+            // Write out the cache file.
+            var stream = File.Create(context.FileListPath);
+            using var writer = new StreamWriter(stream) { AutoFlush = true };
+            writer.WriteLine(string.Join(Environment.NewLine, filePathList));
+            stream.Flush();
+
+            return true;
+        }
+
+        private static string GetDocument(
+            string documentName,
+            string projectName,
+            string outputDirectory,
+            MethodInfo generateMethod,
+            object service)
+        {
+            Reporter.WriteInformation(Resources.FormatRetrievingDocument(documentName));
+
+            var stream = new MemoryStream();
+            using var writer = new StreamWriter(stream) { AutoFlush = true };
+            var resultTask = (Task)InvokeMethod(generateMethod, service, new object[] { documentName, writer });
+            if (resultTask == null)
+            {
+                return null;
+            }
+
+            var finishedIndex = Task.WaitAny(resultTask, Task.Delay(TimeSpan.FromMinutes(1)));
```

Comment:
Could this be simplified using https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=netframework-4.8#System_Threading_Tasks_Task_Wait_System_TimeSpan_